sbin/prepareWorkspace.sh

```diff
@@ -38,14 +37,58 @@ ALSA_LIB_CHECKSUM=${ALSA_LIB_CHECKSUM:-5f2cd274b272cae0d0d111e8a9e363f0878332915
 ALSA_LIB_GPGKEYID=${ALSA_LIB_GPGKEYID:-A6E59C91}
 FREETYPE_FONT_SHARED_OBJECT_FILENAME="libfreetype.so*"
 
+
+copyFromDir() {
+  echo "Copying OpenJDK source from  ${BUILD_CONFIG[OPENJDK_LOCAL_SOURCE_ARCHIVE_ABSPATH]} to $(pwd)/${BUILD_CONFIG[OPENJDK_SOURCE_DIR]} to be built"
+  # We really do not want to use .git for dirs, as we expect user have them set up, ignoring them
+  local files=$(find "${BUILD_CONFIG[OPENJDK_LOCAL_SOURCE_ARCHIVE_ABSPATH]}" -maxdepth 1 -mindepth 1 | grep -v -e "/workspace$" -e "/build$" -e "/.git" -e -"/build/")
+  # SC2086 (info): Double quote to prevent globbing and word splitting.
+  # globbing is intentional here
+  # shellcheck disable=SC2086
+  cp -rf $files "./${BUILD_CONFIG[OPENJDK_SOURCE_DIR]}/"
+}
+
+unpackFromArchive() {
+  echo "Extracting OpenJDK source tarball ${BUILD_CONFIG[OPENJDK_LOCAL_SOURCE_ARCHIVE_ABSPATH]} to $(pwd)/${BUILD_CONFIG[OPENJDK_SOURCE_DIR]} to build the binary"
+  # If the tarball contains .git files, they should be ignored later
+  # todo, support also zips?
+  pushd "./${BUILD_CONFIG[OPENJDK_SOURCE_DIR]}"
+    local topLevelItems=$(tar --exclude='*/*' -tf  "${BUILD_CONFIG[OPENJDK_LOCAL_SOURCE_ARCHIVE_ABSPATH]}" | wc -l)
+    if [ "$topLevelItems" -eq "1" ] ; then
+      echo "Source tarball contains exactly one directory"
+      tar --strip-components 1 -xf "${BUILD_CONFIG[OPENJDK_LOCAL_SOURCE_ARCHIVE_ABSPATH]}"
```

Comment:
But what else to do about it?-( I saw --strip-components on many places over all the repos in adoptium.

Reply:
I've stripped it out from a few places in our code the past - if it's something that only runs on Linux/Windows (cygwin) install then it's ok since it's a reasonably safe assumption they we'll be using GNU tarbut this could be used elsewhere. I've worked around it with something along the lines of this in the past:
```
mkdir tmp
cd tmp && tar --strip-components 1 -xf "${BUILD_CONFIG[OPENJDK_LOCAL_SOURCE_ARCHIVE_ABSPATH]}"
cd .. && mv tmp/*/* .
rmdir tmp
```

Reply:
> I've stripped it out from a few places in our code the past - if it's something that only runs on Linux/Windows (cygwin) install then it's ok since it's a reasonably safe assumption they we'll be using GNU tarbut this could be used elsewhere. I've worked around it with something along the lines of this in the past:
> 
> ```
> mkdir tmp
> cd tmp && tar --strip-components 1 -xf "${BUILD_CONFIG[OPENJDK_LOCAL_SOURCE_ARCHIVE_ABSPATH]}"
> cd .. && mv tmp/*/* .
> rmdir tmp
> ```

You mean 
```
...
 cd tmp && tar  -xf "${BUILD_CONFIG[OPENJDK_LOCAL_SOURCE_ARCHIVE_ABSPATH]}"
...
 ```

thas nice hack.   It would deserve method in commons.